server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -153,10 +153,15 @@ public boolean updateMapping(final IndexMetadata currentIndexMetadata, final Ind
         assert newIndexMetadata.getIndex().equals(index()) : "index mismatch: expected " + index()
             + " but was " + newIndexMetadata.getIndex();
 
+        if (currentIndexMetadata != null && currentIndexMetadata.getMappingVersion() == newIndexMetadata.getMappingVersion()) {
+            assertMappingVersion(currentIndexMetadata, newIndexMetadata, mapper);
```

Comment:
In the referenced commit third param is empty map which was supposed to hit first assertion 
`assert updatedEntries.isEmpty() : updatedEntries;`
https://github.com/elastic/elasticsearch/blob/65f6fb8e94fdff217dcbc59cb1ae68c4c6c40f8c/server/src/main/java/org/elasticsearch/index/mapper/MapperService.java#L293

and in our version (without types map) in a short-curcuit call we need to hit 
` assert updatedMapper == mapper;` so I'm passing current mapper on the first `assertMappingVersion` call. Other changes are aligned.